src/index.c

```diff
@@ -1135,12 +1133,13 @@ static int OI_SkipTo(void *ctx, t_docId docId, RSIndexResult **hit) {
   //  nc->maxDocId, nc->lastDocId);
 
   int found = 0;
-  if (nc->lastDocId > nc->maxDocId) {
-    return INDEXREAD_EOF;
-  }
 
   // Set the current ID
   nc->lastDocId = docId;
```

Comment:
@MeirShpilraien 
By setting `nc->lastDocId = docId` before `if`, we return `INDEXREAD_EOF` instead of an empty `hit`